File: src/types/companyEmployee.ts
// TypeScript interfaces for CompanyEmployee functionality
export interface CompanyEmployee {
  id: number;
  userName?: string;
  militaryNumber?: number;
  fullName: string;
  gradeName?: string;
  createdBy?: string;
  updatedBy?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateCompanyEmployeeRequest {
  userName?: string;
  fullName: string;
  gradeName?: string;
}

export interface UpdateCompanyEmployeeRequest {
  userName?: string;
  fullName: string;
  gradeName?: string;
}

export interface CompanyEmployeeFormData {
  userName: string;
  fullName: string;
  gradeName?: string;
}

// API Response types
export interface CompanyEmployeesResponse {
  success: boolean;
  data: CompanyEmployee[];
  pagination: {
    currentPage: number;
    pageSize: number;
    totalCount: number;
    totalPages: number;
  };
  error?: string;
}

export interface CompanyEmployeeResponse {
  success: boolean;
  data: CompanyEmployee;
  error?: string;
}

File: pma-api-server/src/PMA.Core/Entities/CompanyEmployee.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PMA.Core.Entities;

[Table("CompanyEmployees")]
public class CompanyEmployee
{
    [Key]
    public int Id { get; set; }

    [MaxLength(500)]
    public string? UserName { get; set; }

    public int? MilitaryNumber { get; set; }

    [Required]
    [MaxLength(500)]
    public string FullName { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? GradeName { get; set; }

    [MaxLength(500)]
    public string? CreatedBy { get; set; }

    [MaxLength(500)]
    public string? UpdatedBy { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }
}